deps/hiredis/sds.c

```diff
@@ -990,7 +990,7 @@ hisds *hi_sdssplitargs(const char *line, int *argc) {
                         current = hi_sdscatlen(current,p,1);
                     }
                 } else if (insq) {
-                    if (*p == '\\' && *(p+1) == '\'') {
+                    if (*p == '\\' && *(p+1) == '\'' && *(p+2)) {
```

Comment:
@yossigo This is not enough. It only solves the last argument.
```
127.0.0.1:6383> mset hello 'foo\' world 'bar\'
Invalid argument(s)
```
```suggestion
                    if (*p == '\\' && *(p+1) == '\'' && *(p+2) && *(p+2) != ' ' &&
                            *(p+2) != '\n' && *(p+2) != '\r' && *(p+2) != '\t') {

```

Reply:
This is dangerous territory. Now, how can we tell the difference between the end of a string and an escaped apostrophe in the end of a word?

    mset hello 'we better watch what we\'re doin\' and 'where we\'re goin\' don\'t we?'

Reply:
@zuiderkwast  My bad. You are right. Maybe this issue can't be fixed.

Reply:
I suppose it can't be fixed without escaping the backslash.

Accepting `\'` in the end of the command line is safe, since it only affects what's otherwise invalid arguments, but please @yossigo let's make the breaking change in 7.0 and allow escaping the escape character so you can write `mset hello 'foo\\' world 'bar\\'`. It's the behaviour that makes sense and what people would expect.

Reply:
@huangzhw You are right of course, I forgot we handle multiple single quotes (was apparently too late in the day). 
@zuiderkwast I agree it's dangerous/smelly but the way I see it we have only three options here:
1. Leave it as a known issue and not fix it.
2. Do the awkward fix.
3. Break compatibility.

Reply:
OK, this (awkward) fix is fine, but I just want to highlight that treating all occurrences `\'` followed by space or newline would be a breaking change. That's an escaped quote only.

To be clear, `'doin\' great'` should be interpreted as `"doin' great"`, not backslash included.

Doing 2 doesn't exclude also doing 3; they're safe to combine. Why not do 2 for Redis 6.2 and 3 for Redis 7.0? (Because double-quoting is recommended instead?)

Reply:
@zuiderkwast  I think fix 2 is unnecessary. It can only fix half not all. This is just a tiny bug which we always can use double quotes. So can't fix totally is unnecessary.